scripts/import_libcbor.py

```diff
@@ -0,0 +1,139 @@
+import os
+import tempfile
+import shutil
+import subprocess
+import argparse
+
+def parse_version(version_string):
+    if version_string.startswith('v'):
+        version_string = version_string[1:]
+    parts = version_string.split('.')
+    if not parts:
+        raise ValueError("Invalid version string")
+    major = int(parts[0])
+    minor = 0
+    patch = 0
+    if len(parts) > 1:
+        minor = int(parts[1])
+
+    if len(parts) > 2:
+        patch = int(parts[2])
+
+    return major, minor, patch
+
+argument_parser = argparse.ArgumentParser(
+    description="Helper to import libcbor as external dependency.")
+
+argument_parser.add_argument("--version",
+                                default="v0.11.0", help="Version string to import")
+
+args = argument_parser.parse_args()
+major_version, minor_version, patch_version = parse_version(args.version)
+
+GENERATED_NOTES = """/**
+ * DO NOT DIRECTLY MODIFY THIS FILE:
+ *
+ * The code in this file is generated from scripts/import_libcbor.py
+ * and any modifications should be in there.
+ */
+"""
+
+CBOR_EXPORT_H = """
+#ifndef CBOR_EXPORT_H
+#define CBOR_EXPORT_H
+
+/* Don't export anything from libcbor */
+#define CBOR_EXPORT
+
+#endif /* CBOR_EXPORT_H */
+"""
+
+CONFIGURATION_H = f"""
+#ifndef LIBCBOR_CONFIGURATION_H
+#define LIBCBOR_CONFIGURATION_H
+
+#define CBOR_MAJOR_VERSION {major_version}
+#define CBOR_MINOR_VERSION {minor_version}
+#define CBOR_PATCH_VERSION {patch_version}
+
+#define CBOR_BUFFER_GROWTH 2
+#define CBOR_MAX_STACK_SIZE 2048
+#define CBOR_PRETTY_PRINTER 1
+
+#if defined(_MSC_VER)
+#    define CBOR_RESTRICT_SPECIFIER
+#else
+#    define CBOR_RESTRICT_SPECIFIER restrict
+#endif
+
+#define CBOR_INLINE_SPECIFIER
+
+/* Ignore the compiler warnings for libcbor. */
+#ifdef _MSC_VER
+#    pragma warning(disable : 4028)
+#    pragma warning(disable : 4715)
+#    pragma warning(disable : 4232)
+#    pragma warning(disable : 4068)
+#    pragma warning(disable : 4244)
+#    pragma warning(disable : 4701)
+#    pragma warning(disable : 4703)
+#endif
+
+#ifdef __clang__
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wreturn-type"
+#elif defined(__GNUC__)
+#    pragma GCC diagnostic push
```

Comment:
trivial: I don't think there's any point in "pushing" a new level in the warning stack, since we never "pop" it off
```suggestion
#    pragma clang diagnostic ignored "-Wreturn-type"
#elif defined(__GNUC__)
```